src/test/scala/chiselTests/TypeAliasSpec.scala

```diff
@@ -0,0 +1,321 @@
+
+      w :#= io.in
+      io.out :#= w
+    }
+
+    val chirrtl = ChiselStage.emitCHIRRTL(new Test)
+
+    chirrtl should include("type FooBundle = { x : UInt<8>, y : UInt<8>}")
+    chirrtl should include("flip in : FooBundle")
+    chirrtl should include("out : FooBundle")
+    chirrtl should include("wire w : FooBundle")
+  }
+
+  "Bundles with opt-in FIRRTL type aliases" should "support nesting" in {
+    class Test extends Module {
+      class BarBundle extends Bundle {
+        override def aliasName = Some(BundleAlias("Buzz"))
+
+        val y = UInt(8.W)
+      }
+      class FooBundle extends Bundle {
+        override def aliasName = Some(BundleAlias("Fizz"))
+
+        val x = UInt(8.W)
+        val bar = new BarBundle
+      }
+
+      val io = IO(new Bundle {
+        val in = Input(new FooBundle)
+        val out = Output(new FooBundle)
+      })
+
+      val w = Wire(new FooBundle)
+
+      w :#= io.in
+      io.out :#= w
+    }
+
+    val chirrtl = ChiselStage.emitCHIRRTL(new Test)
+
+    chirrtl should include("type Fizz = { x : UInt<8>, bar : Buzz}")
+    chirrtl should include("flip in : Fizz")
+    chirrtl should include("out : Fizz")
+    chirrtl should include("wire w : Fizz")
+  }
+
+  "Bundles with opt-in FIRRTL type aliases" should "generate normal aliases for coerced, monodirectional/unflipped bundles" in {
+    class MonoStrippedTest extends Module {
+      class BarBundle extends Bundle {
+        override def aliasName = Some(BundleAlias("Buzz"))
+
+        val x = UInt(8.W)
+        val y = UInt(8.W)
+      }
+
+      val io = IO(new Bundle {
+        val in = Input(new BarBundle)
+        val out = Output(new BarBundle)
+      })
+
+      io.out :#= io.in
+    }
+
+    val chirrtl = ChiselStage.emitCHIRRTL(new MonoStrippedTest)
+    // Alias for the bundle since it wouldn't be stripped
+    chirrtl should include("type Buzz = { x : UInt<8>, y : UInt<8>}")
+  }
+
+  "Bundles with opt-in FIRRTL type aliases" should "generate modified aliases for coerced, bidirectional bundles" in {
+    class StandardStrippedTest extends Module {
+      class BarBundle extends Bundle {
+        override def aliasName = Some(BundleAlias("Buzz"))
+
+        val x = UInt(8.W)
+        // This flip gets stripped by both Input() and Output()
+        val y = Flipped(UInt(8.W))
+      }
+
+      val io = IO(new Bundle {
+        val in = Input(new BarBundle)
+        val out = Output(new BarBundle)
+      })
+
+      io.out :#= io.in
+    }
+
+    val chirrtl = ChiselStage.emitCHIRRTL(new StandardStrippedTest)
+    // No unmodified alias for the bidirectional bundle since it isn't actually bound
+    chirrtl shouldNot include("type Buzz = { x : UInt<8>, flip y : UInt<8>}")
+    // No unmodified alias for the stripped bundle
+    chirrtl shouldNot include("type Buzz = { x : UInt<8>, y : UInt<8>}")
+    // Modified alias for the stripped bundle
+    chirrtl should include("type Buzz_stripped = { x : UInt<8>, y : UInt<8>}")
```

Comment:
let's chat about this algorithm -- we should discuss maybe making this `_stripped` suffix something also in the user's control

Reply:
It is under the user's control now